packages/network/lib/connect.ts

```diff
@@ -7,24 +7,29 @@ import tls from 'tls'
 
 const debug = debugModule('cypress:network:connect')
 
-export function byPortAndAddress (port: number, address: net.Address) {
+export function byPortAndAddress (hostname: string, port: number, address: net.Address) {
   // https://nodejs.org/api/net.html#net_net_connect_port_host_connectlistener
   return new Bluebird<net.Address>((resolve, reject) => {
     const onConnect = () => {
-      client.end()
+      client.destroy()
       resolve(address)
     }
 
-    const client = net.connect(port, address.address, onConnect)
+    let client
+
+    if (port === 443) {
```

Comment:
This won't cover all instances of HTTPS usage, there needs to be another signal, HTTPS requests can happen on ports other than 443.

Reply:
Yep was thinking to use protocol than port. Just wanted to confirm the approach and will change the code upon the agreement.